Marlin/ultralcd.cpp

```diff
@@ -2741,7 +2741,15 @@ void lcd_update() {
 void set_utf_strlen(char* s, uint8_t n) {
   uint8_t i = 0, j = 0;
   while (s[i] && (j < n)) {
-    if ((s[i] & 0xC0u) != 0x80u) j++;
+    #if ENABLED(MAPPER_NON)
+      #if ENABLED(DISPLAY_CHARSET_ISO10646_KANA)
+        if (s[i] != 0xDEu) j++;
+      #else
+        j++;
+      #endif
+    #else
+      if ((s[i] & 0xC0u) != 0x80u) j++;
+    #endif
```

Comment:
DE is a character at its own here.
We are pointing directly into the charset.
No combining of the accent and the according character can be made.
```
-    if ((s[i] & 0xC0u) != 0x80u) j++;
 +    #if ENABLED(MAPPER_NON)
 +        j++;
 +    #else
 +      if ((s[i] & 0xC0u) != 0x80u) j++;
 +    #endif
```
should be more correct.